modules/rpc/stats/metrics.go

```diff
@@ -18,19 +18,39 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package modules
+package stats
+
+import "github.com/uber-go/tally"
 
 const (
 	//TagModule is module tag for metrics
 	TagModule = "module"
 	// TagType is either request or response
 	TagType = "type"
-	// TagStatus is status of the request
-	TagStatus = "status"
+	// TagProcedure is the procedure name
+	TagProcedure = "procedure"
+	//TagMiddleware is middleware type
+	TagMiddleware = "middleware"
 )
 
 // HTTPTags creates metrics scope with defined tags
-var HTTPTags = map[string]string{
-	TagModule: "http",
-	TagType:   "request",
+var (
+
+	// RPCTags creates metrics scope with defined tags
+	RPCTags = map[string]string{
+		TagModule: "rpc",
+		TagType:   "request",
+	}
+
+	// RPCAuthFailCounter counts auth failures
+	RPCAuthFailCounter tally.Counter
+	// RPCHandleTimer is a turnaround time for rpc handler
+	RPCHandleTimer tally.Scope
+)
+
+// SetupRPCMetrics allocates counters for necessary setup
+func SetupRPCMetrics(scope tally.Scope) {
+	rpcTagsScope := scope.Tagged(RPCTags)
+	RPCAuthFailCounter = rpcTagsScope.Tagged(map[string]string{TagMiddleware: "auth"}).Counter("rpc.auth.fail")
```

Comment:
you lose the group/alias/apply by node functions when you use m3ql. if you have the tags you can aggregate by tags instead.